app/pages/list/list.js

```diff
@@ -0,0 +1,34 @@
+import {IonicApp, Page, NavController, NavParams} from 'ionic/ionic';
+
+@Page({
+  templateUrl: 'build/pages/list/list.html'
+})
+export class ListPage {
+  constructor(app: IonicApp, nav: NavController, navParams: NavParams) {
+    this.nav = nav;
+
+    // If we navigated to this page, we will have an item available as a nav param
+    this.selectedItem = navParams.get('item');
+
+    this.icons = ['flask', 'wifi', 'beer', 'football', 'basketball', 'paper-plane',
+    'american-football', 'boat', 'bluetooth', 'build'];
+
+    this.items = [];
+    for(let i = 1; i < 11; i++) {
+      this.items.push({
+        title: 'Item ' + i,
+        note: 'This is item #' + i,
+        icon: this.icons[Math.floor(Math.random() * this.icons.length)]
+      });
+    }
+  }
+
+  itemTapped(event, item) {
+
+    console.log('You selected:', item.title);
+
+      this.nav.push(ItemDetailsPage, {
+        item: item
+      });
```

Comment:
In Ionic 2, if you want to add a new view you push it onto the navigation stack, and if you want to remove it you pop it. This is like pushing and popping an array, or if you’re unfamiliar with that imagine a stack of papers. The first piece of paper is your rootPage, then you can push a new page by placing a piece of paper on top of that – now you can only see the top paper. You can continue pushing more and more pieces of paper on top of the previous pieces creating a stack. Then you can easily navigate back through that stack by simply removing the top piece of paper (but you wouldn’t remove a piece of paper from the middle of the stack). You keep removing pieces of paper until you are back where you want to be.

In the ListPage component, we push the ItemDetailsPage with the itemTapped function (which is triggered from the ListPage template, and takes in a particular item that was tapped).
This will push the ItemDetailsPage component onto the navigation stack, making it the active view, and it passes in the details of the item that was clicked.